src/main/java/com/google/api/codegen/DiscoveryImporter.java

```diff
@@ -453,11 +469,19 @@ private String responseTypeUrl(JsonNode root) {
    * format, if exists, is recorded in {@link ApiaryConfig#stringFormat}.
    */
   private Field.Kind getFieldKind(
-      String type, String field, String kindName, JsonNode formatNode, JsonNode patternNode) {
+      String type,
+      String field,
+      String kindName,
+      JsonNode formatNode,
+      JsonNode patternNode,
+      JsonNode locationNode) {
     String format = "";
     if (formatNode != null) {
       format = formatNode.asText();
     }
+    if (locationNode != null) {
+      config.getFieldLocation().put(type, field, locationNode.asText());
```

Comment:
It might be more robust to establish an enum for the two options we know about, just in case there are options we don't know about.